### DISCUSSION TITLE
Coc API

### DISCUSSION THREAD

**Discussion by Cris-lml007** [neoclide/coc.nvim#4936] [General]
Hi, I want to make a plugin for coc that supports `Blazor`, `Razor`; I already have the idea of how to do it but I don't know how to implement it, I would like to retrieve the alerts that the `lsp` sends to the buffer but I can't find a way to do it or it is not implemented in the coc api. I would like you to help me with this information or to implement it in an API please.

**Comment by fannheyward** [comment]
There's no full API docs, here's some resources you can use:

1. `:h coc-api-extension` for a quick guide, and `:h coc-api.txt`
2. coc.nvim provides similar API as VSCode https://github.com/neoclide/coc.nvim/blob/master/typings/index.d.ts, if there's an extension for Blazor, you can _clone_ it to coc.nvim side
3. check other extension's implementation
